e2e/cypress/integration/api/iam/02_policies_api.spec.ts

```diff
@@ -87,11 +110,359 @@ describeIfIAMV2('policies API', () => {
         body: {
           name: `${cypressPrefix} policy ${now}`,
           members: ['team:local:test'],
-          projects: [projectID]
+          projects: [project1.id]
         }
       }).then((response) => {
        expect(response.status).to.equal(400);
       });
     });
   });
+
+  describeIfIAMV2p1('project assignment enforcement', () => {
+    let nonAdminToken = '';
+    const nonAdminTokenID = `${cypressPrefix}-nonadmin-token-${now}`;
+    const statementProjects = [project1.id];
+    const policyID = `${cypressPrefix}-policy-${now}`;
+
+    before(() => {
+      cy.request({
+        auth: { bearer: adminToken },
+        method: 'POST',
+        url: '/apis/iam/v2beta/tokens',
+        body: {
+            active: true,
+            id: nonAdminTokenID,
+            name: 'Nonadmin Token'
+        }
+      }).then((response) => {
+          nonAdminToken = response.body.token.value;
+      });
+
+      cy.request({
+        auth: { bearer: adminToken },
+        method: 'POST',
+        url: '/apis/iam/v2beta/policies',
+        body: {
+            name: 'token-access',
+            id: `${cypressPrefix}-token-access-policy-${now}`,
+            members: [ `token:${nonAdminTokenID}` ],
+            statements: [
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:policies:*' ],
+                projects: [ '*' ]
+              },
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:projects:assign' ],
+                projects: [ project1.id ]
+              }
+            ],
+            projects: []
+        }
+      });
+    });
+
+    after(() => {
+      cy.cleanupProjectsByIDPrefix(adminToken, cypressPrefix);
+      cy.cleanupPoliciesByIDPrefix(adminToken, cypressPrefix);
+      cy.cleanupTokensByIDPrefix(adminToken, cypressPrefix);
+    });
+
+    beforeEach(() => {
+      cy.request({
+        auth: { bearer: adminToken },
+        method: 'DELETE',
+        url: `/apis/iam/v2beta/policies/${policyID}`,
+        failOnStatusCode: false
+      });
+    });
+
+    afterEach(() => {
+      cy.request({
+        auth: { bearer: adminToken },
+        method: 'DELETE',
+        url: `/apis/iam/v2beta/policies/${policyID}`,
+        failOnStatusCode: false
+      });
+    });
+
+    describe('POST /apis/iam/v2beta/policies', () => {
+      it('admin can create a new policy with no projects', () => {
+        cy.request({
+            auth: { bearer: adminToken },
```

Comment:
🤔 This block is *almost* repeated many times here -- i.e., there's most of the time just one nuance changed. I wonder if we could come up with a default request object, like
```typescript
const req = {
  auth: { bearer: adminToken },
  method: 'GET',
  url: '/apis/iam/v2beta/policies'
};
```
and use this, with the one-changed-thing updated, below, e.g.:
```typescript
cy.request({ ...req, method: 'POST', body: policyWithProjects(policyID, [], statementProjects) });
```
```typescript
cy.request({ ...req, method: 'DELETE', url: `/apis/iam/v2beta/policies/${policyID}) });
```
etc.

What do you think?

Reply:
It's a bit different for the non-admin case because we aren't using a bearer token but we can probably dedup a bit.

Reply:
Done